walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,77 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {  
+        
+    //Assigning duration of the meeting 
+     long meetingDuration = request.getDuration();
+
+
+    //Saving collection of attendees 
+    Collection<String> meetingAttendees = request.getAttendees();
+
+    //Finding all incompatible time ranges
+   ArrayList<TimeRange> incompatibleRanges = findIncompatibleRanges(events, meetingAttendees);
+
+     
+    //Sorting the range by chronological order 
+    Collections.sort(incompatibleRanges, TimeRange.ORDER_BY_START);
+    Collection<TimeRange> availableRanges = new ArrayList<TimeRange>();
+
+
+    int meetingRequestStartTime = 0; 
+    for (TimeRange eventRange : incompatibleRanges){
+      int rangeEnd = eventRange.end();
+      int rangeStart = eventRange.start();
+        if (!(meetingRequestStartTime + meetingDuration > rangeStart)){ 
```

Comment:
nit: This would probably be easier to read if written as
`if(meetingRequestStartTime + meetingDuration <= rangeStart)`